zips/zip-0302.rst

```diff
@@ -64,6 +67,57 @@ contents of a memo. It does not define consensus requirements.
   assumption about the memo. This value was used ambiguously in the past by private
   agreement; applications SHOULD prefer ``0xFF`` which is unambiguously for this purpose.
 
+Structured memos
+----------------
+
+A memo can be used to store structured data by setting its first byte to a value of
+``0xF7``. The remaining bytes contain a set of "memo parts", encoded as concatenated
+Type-Length-Value tuples, followed by fixed padding (if any memo bytes remain after the
```

Comment:
In a pairing, @nuttycom and I decided that this needed to be Type-Version-Length-Value, so that individual part types can have their value encodings updated, while still enabling the "only one of each part type" to be enforced by wallets. This also encourages part type specifiers to version their protocols.